### PR TITLE
Remove theme directory confirmation during tests and make confirmation dialogue respect `SHOPIFY_CLI_TTY`

### LINKED ISSUE #1136: `theme pull` in a non-empty and non-theme directory hangs on CI environment (even with `SHOPIFY_CLI_TTY`)
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme

### Expected behavior

`shopify theme pull` being run in a non-theme directory and in an CI/CD environment (→ [docs](https://shopify.dev/themes/tools/cli/ci-cd#example-github-actions)) should display confirmation dialogue _It doesn’t seem like you’re running this command in a theme directory. Are you sure you want to proceed?_ and not hang if `SHOPIFY_CLI_TTY` is set to `0`.

### Actual behavior

`theme pull` hangs, never completes, and displays no further message.

### Stack trace

```shell
[STARTED] Installing theme dependencies
2023-01-18T11:19:35.391Z: 
Running system process:
  · Command: ruby -v
  · Working directory: /home/runner/work/shopify-theme/shopify-theme

2023-01-18T11:19:35.417Z: 
Running system process:
  · Command: bundle -v
  · Working directory: /home/runner/work/shopify-theme/shopify-

…

2023-01-18T11:19:35.747Z: 
Running system process:
  · Command: bundle install
  · Working directory: /home/runner/.cache/shopify-cli-nodejs/vendor/ruby-cli/2.34.0

[SUCCESS] Installing theme dependencies
2023-01-18T11:19:48.295Z: 
Running system process:
  · Command: bundle exec shopify theme pull /home/runner/work/shopify-theme/shopify-theme --theme 140280135974
  · Working directory: /home/runner/work/shopify-theme/shopify-theme
```

### Reproduction steps

Create a github actions file with following settings

```yaml
name: Theme deploy

on: [push]

jobs:
  deploy:
    name: Deploy
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - uses: actions/setup-node@v3
        with:
          node-version: 18
      - uses: ruby/setup-ruby@v1
        with:
          ruby-version: 3.1
          bundler: 'latest'
      - name: Install Shopify CLI
        run: npm install -g @shopify/cli @shopify/theme
      - name: Deploy
        env:
          # Store URL, like your-store.myshopify.com
          SHOPIFY_FLAG_STORE: '${{ secrets.SHOPIFY_STORE }}'
          # Password generated from Theme Access app
          SHOPIFY_CLI_THEME_TOKEN: ${{ secrets.SHOPIFY_CLI_THEME_TOKEN }}
          SHOPIFY_CLI_TTY: 0
        run: shopify theme push --path ./path/to/your/theme --theme your-theme-name-or-id
```

Populate the secrets
and attempt to deploy

### Operating System

Github Actions 

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.34

### Shell

_No response_

### Node version (run `node -v` if you're not sure)

_No response_

### What language and version are you using in your application?

_No response_

### PR DESCRIPTION
**Pull request by Poitrin**
### WHY are these changes introduced?

Fixes #1136 

#### Issue 1: `@ctx.testing?` is `true` even outside a testing environment

CLI was hanging when `theme pull` was executed in a directory with already existing files.
CLI should have shown question

> It doesn’t seem like you’re running this command in a theme directory. Are you sure you want to proceed?

but was never shown, due to

```ruby
raise "Current theme directory can't be confirmed during tests" if @ctx.testing?
```

Discussion why `raise … if @ctx.testing?` was introduced can be found [here](https://github.com/Shopify/shopify-cli/pull/2680#discussion_r1018371466). Because `@ctx.testing?` returns `true` based on `ENV["CI"]`, we can’t differentiate between CLI running in partner’s pipeline or in CLI tests.

#### Issue 2: Confirmation dialogue ignores environment variable `SHOPIFY_CLI_TTY`

Moreover, even after having removed `raise … if @ctx.testing?` and `SHOPIFY_CLI_TTY` being set to `0`, CLI would still hang at the “It doesn’t seem like …” question.

### WHAT is this pull request doing?

1. Removes `raise … if @ctx.testing?`.
2. Shows a warning `It doesn’t seem like you’re running this command in a theme directory.` when `SHOPIFY_CLI_TTY` is set to `0`, but continues to `pull`.
    * `--force` flag as a way to confirm the command will not be included in the error message yet.
      * → https://github.com/Shopify/shopify-cli/pull/2680#discussion_r1017721847
      * → https://github.com/Shopify/cli/issues/760

### How to test your changes?

1. `mkdir ./test-folder`
2. `cd ./test-folder`
4. `touch randomfile1.txt randomfile2.log`
5. `cd ..`
6. Retrieve a random theme ID from `shopify theme list`
7. Run `CI=true pnpm shopify:run theme pull --theme <Your Theme ID> --path="./test-folder"`.
    * Verify that warning `It doesn’t seem like you’re running this command in a theme directory.` and confirmation dialogue `Are you sure you want to proceed?` are shown.
    * `Ctrl+C`.
9. Run `SHOPIFY_CLI_TTY=0 CI=true pnpm shopify:run theme pull --theme <Your Theme ID> --path="./test-folder"`.
    * Verify that CLI displays warning `It doesn’t seem like you’re running this command in a theme directory.` but continues to `pull`.
    * `Ctrl+C` the pull and make sure that your test-folder is a non-theme directory (again).
11. Run `CI=true pnpm shopify:run theme pull --theme <Your Theme ID> --path="./test-folder" --force`.
    * Verify that neither warning nor confirmation dialogue are shown and that theme files get pulled.

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/cli-kit/assets/cli-ruby/lib/project_types/theme/commands/common/root_helper.rb [hunk 1]
```diff
@@ -49,12 +49,14 @@ def exist_and_not_empty?(root)
         private
 
         def current_directory_confirmed?
-          raise "Current theme directory can't be confirmed during tests" if @ctx.testing?
+          message = @ctx.message("theme.current_directory_is_not_theme_directory")
+          @ctx.abort(message) unless ShopifyCLI::Environment.interactive?
 
+          message += " " + @ctx.message("theme.confirm_current_directory")
```
Comment (mgmanzella) [inline packages/cli-kit/assets/cli-ruby/lib/project_types/theme/commands/common/root_helper.rb:55]: /supernit, this is more idiomatic plus we're explicitly showing we're adding a new line 

```suggestion
          message += " \n#{@ctx.message('theme.confirm_current_directory')}"
```

**Review by mgmanzella — APPROVED** [review]
noice 🔥 thank you @Poitrin!

**Comment by Poitrin** [comment]
@karreiro suggested to replace the `abort` with a `warn`, which sounds like a better approach, especially as the error message does not show any way to `--force` the `pull`.
@mgmanzella – can you take a 2ⁿᵈ look again? :)

**Review by karreiro — APPROVED** [review]
Thank you, @Poitrin!

**Review by mgmanzella — APPROVED** [review]

**Pull request merged**